### ISSUE TITLE
Popup not working.

### ISSUE THREAD

**Issue by ForgivenNin** [slogsdon/javascript-browser-contexts#1]
Hey!

I was really interested to see how this repository relates to the parent domain/window problem, but for some reason when I click either of the buttons on the `example` HTML, the event listener is not going off. I gave the JavaScript a quick glance to find that you seem to have done everything right. However, I could not put my finger on the root of the issue.

**Comment by slogsdon** [comment]
@ForgivenNin To make sure I'm looking at this correctly, are parent window and child window on the same origin (scheme + domain + port) or different origins?

**Comment by slogsdon** [comment]
@ForgivenNin I tested a few things, but I'm not seeing any problems with same origin nor cross origin. Screenshot:

<img width="1680" alt="Screen Shot 2020-03-24 at 10 22 55 PM" src="https://user-images.githubusercontent.com/647624/77495175-34ac4200-6e1e-11ea-8206-0f0c56dfafba.png">

In this, I've loaded the parent page at `http://localhost:7777/example/` and clicked the "Create iFrame" button. This creates a hidden iframe with pointing at `http://localhost:8989/example/child.html`.

In the console, you should see two message events to the parent window (prefixed with "parent window"): 1) that is the parent window's own load event that the library is currently bubbling up as a message and 2) the child iframe window's load event. The child's load event has the correct origin (`http://localhost:8989`) and the correct id (`ZDRkMjU3ZWQtYzMxNy0wYzVkLThiNzQtNTdiMDQ3YmNjNzU5KQ==`) that's generated by the library.

Following these two log messages, there's a `console.log` from the example showing all created windows:

https://github.com/slogsdon/javascript-windows/blob/8ff67679eb3368615ea294e41e5002daa35058de/example/index.js#L25

Finally, there's a fourth log from the child window receiving a message from the parent. This has the correct origin (`http://localhost:7777`).

I updated the example to better show a distinction between parent and child with separate HTML as well as show how windows can be nested (e.g. iframe within an iframe, iframe creating a popup). Can you try the latest example code on your end and let me know what you're seeing?

**Comment by ForgivenNin** [comment]
I tried it on Opera and then it gave me the CORS error, so I went ahead and moved it to Safari, where I can disable such restrictions. I disabled both CORS and local file restrictions, and this is what I got as soon as the page loaded (and remained this after clicking the buttons).

<img width="372" alt="Screen Shot 2020-03-25 at 11 43 29 AM" src="https://user-images.githubusercontent.com/46826941/77555806-0fefb300-6e8e-11ea-9b3e-d8effaf6d63b.png">

**Comment by slogsdon** [comment]
Thanks for the additional info! The screenshot from Safari was insightful, so I pushed a [small commit](https://github.com/slogsdon/javascript-browser-contexts/commit/47c9af9999f7e3295dbe3246ad71c25d19e22140) to fix this as Safari doesn't like having an undefined value for the origin which should have only affected the initial load.

As far as CORS, you shouldn't need to disable anything as long as you're hitting the example or your own code via HTTP/HTTPS. Loading them as files (`file:///....`) will cause you problems. I tested against the Chromium based browsers (Chrome, Opera, and Edge) as well as Safari and Firefox, and all of them are working as expected with no modifications to default browser settings.
